app/src/utils/api.test.tsx

```diff
@@ -0,0 +1,177 @@
+        uuid2: "45",
+      });
+    });
+  });
+});
+
+describe("apiResponse", () => {
+  const mockApiUrl = "mockurl.org";
+  const mockAuthToken = "mockAuthToken";
+
+  beforeAll(() => {
+    process.env.AUTH_TOKEN = mockAuthToken;
+  });
+
+  it("makes a GET request and returns response", async () => {
+    const mockResponse = { nyplAPI: { response: "mockResponse" } };
+    (global as any).fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        status: 200,
+        json: async () => mockResponse,
+      })
+    ) as jest.Mock;
+
+    const response = await apiResponse(mockApiUrl);
+    expect(fetch).toHaveBeenCalledWith(mockApiUrl, {
+      method: "GET",
+      headers: {
+        Authorization: `Token token=${mockAuthToken}`,
+      },
+      body: undefined,
+    });
+    expect(response).toEqual("mockResponse");
+  });
+
+  it("makes a POST request with a body and returns response", async () => {
+    const mockBody = { key: "value" };
+    const mockResponse = { result: "mockPostResponse" };
+    (global as any).fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        status: 200,
+        json: async () => mockResponse,
+      })
+    ) as jest.Mock;
+
+    const response = await apiResponse(mockApiUrl, {
+      method: "POST",
+      body: mockBody,
+    });
+
+    expect(fetch).toHaveBeenCalledWith(mockApiUrl, {
+      method: "POST",
+      headers: {
+        Authorization: `Token token=${mockAuthToken}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(mockBody),
+    });
+    expect(response).toEqual(mockResponse);
+  });
+
+  it("adds query parameters for GET requests", async () => {
+    const mockParams = { param1: "value1", param2: "value2" };
+    const mockResponse = { nyplAPI: { response: "mockGetResponseWithParams" } };
+    (global as any).fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        status: 200,
+        json: async () => mockResponse,
+      })
+    ) as jest.Mock;
+
+    const response = await apiResponse(mockApiUrl, {
+      params: mockParams,
+    });
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${mockApiUrl}?param1=value1&param2=value2`,
+      {
+        method: "GET",
+        headers: {
+          Authorization: `Token token=${mockAuthToken}`,
+        },
+        body: undefined,
+      }
+    );
+    expect(response).toEqual("mockGetResponseWithParams");
+  });
+
+  it("throws an error for non-200 status", async () => {
+    (global as any).fetch = jest.fn(() =>
+      Promise.resolve({
+        status: 500,
+        statusText: "Internal Server Error",
+      })
+    ) as jest.Mock;
+
+    await expect(apiResponse(mockApiUrl)).rejects.toEqual(
+      new Error("apiResponse: 500 Internal Server Error")
+    );
+  });
+
+  it("throws a timeout error if the request takes too long", async () => {
+    jest.useFakeTimers();
+
+    (fetch as jest.Mock).mockImplementation(() => new Promise(() => {}));
```

Comment:
I was trying to run this test in isolation (using `it.only`) but it fails with the following:

<img width="610" alt="Screenshot 2024-11-05 at 8 19 43 PM" src="https://github.com/user-attachments/assets/835b9323-b259-4ce4-ab4e-b16800776761">

Follow the pattern from the previous test to mock fetch. Ideally, tests should not rely on global mocks or vars.